dispatcher/backend/src/routes/utils.py

```diff
@@ -87,19 +87,24 @@ def remove_secret_flag_from_command(
     ] = f'--{field_name}="{SECRET_REPLACEMENT}"'
 
 
-def remove_upload_secrets(response: dict):
-    """remove keyId and secretAccessKey upload_uri, since we upload logs to
-    S3 but still need the rest of the URL to download scraper logs"""
-    if not has_dict_sub_key(response, ["upload", "logs", "upload_uri"]):
-        return
-    url = urlparse(response["upload"]["logs"]["upload_uri"])
-    response["upload"]["logs"]["upload_uri"] = url._replace(
-        query="&".join(
-            [
-                param
-                for param in url.query.split("&")
-                if not param.lower().startswith("keyid")
-                and not param.lower().startswith("secretaccesskey")
-            ]
-        )
-    ).geturl()
+def remove_s3_secrets(response: dict):
+    """remove keyId and secretAccessKey query params from any URL we might find"""
+    for key in response.keys():
+        if not response[key]:
+            continue
+        if isinstance(response[key], dict):
+            remove_s3_secrets(response[key])
+        else:
+            if not isinstance(response[key], str) or "://" not in response[key]:
+                continue
+            url = urlparse(response[key])
+            response[key] = url._replace(
+                query="&".join(
+                    [
+                        param
+                        for param in url.query.split("&")
```

Comment:
😳 use `parse_qs`

Reply:
🤦🏻 this is already running in production, what a shame ; every time I read this code I had a bad feeling but always said in my mind "this has been reviewed, there was probably nothing best to do, do not loose your time on checking this again". I'll fix it right now!